src/ebay-toggle-button/toggle-button.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { FC } from 'react'
+import cx from 'classnames'
+import { EbayIcon } from '../ebay-icon'
+import { ToggleButtonProps } from './types'
+
+const ToggleButton: FC<ToggleButtonProps> = ({
+    pressed,
+    children,
+    className,
+    title,
+    subtitle,
+    layoutType = 'minimal',
+    icon,
+    img,
+    onToggle,
+    ...rest
+}) => {
+    const handleClick = (e: React.MouseEvent) => {
+        onToggle({ originalEvent: e, pressed: !pressed })
```

Comment:
On React we use the react (and dom) way of sending events. So it will be `onToggle(event, data)`. We have a helper type that you can use https://github.com/eBay/ebayui-core-react/blob/main/src/common/event-utils/types.ts